integration/delete_compat_test.go

```diff
@@ -41,18 +41,82 @@ func TestDeleteCompat(t *testing.T) {
 			filters:    []bson.D{},
 			resultType: emptyResult,
 		},
+		"OrderedTrue": {
+			filters: []bson.D{
+				{{"_id", "string"}},
+				{{"v", bson.D{{"$all", 9}}}},
+				{{"_id", "double"}},
+			},
+			ordered: true,
+		},
+		"OrderedFalse": {
```

Comment:
I think that's already covered by the "Two" test case. Please check if there are other duplicates